packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -21,18 +21,35 @@ import 'template.dart';
 ///
 /// This defines interfaces common to iOS and macOS projects.
 abstract class XcodeBasedProject extends FlutterProjectPlatform  {
-  static const String _hostAppProjectName = 'Runner';
+  late final String hostAppProjectName = () {
+    const String defaultHostAppProjectName = 'Runner';
+    List<FileSystemEntity> contents;
+    if (!hostAppRoot.existsSync()) {
+      return defaultHostAppProjectName;
+    }
+    contents = hostAppRoot.listSync();
+    final FileSystem fileSystem = hostAppRoot.fileSystem;
+    for (final Directory entity in contents.whereType<Directory>()) {
+      // On certain volume types, there is sometimes a stray `._Runner.xcworkspace` file.
+      // Find the first non-hidden xcworkspace and return the directory.
+      if (fileSystem.path.extension(entity.path) == '.xcodeproj' &&
+          !fileSystem.path.basename(entity.path).startsWith('.')) {
+        return fileSystem.path.basenameWithoutExtension(entity.path);
+      }
+    }
+    return defaultHostAppProjectName;
+  }();
 
   /// The parent of this project.
   FlutterProject get parent;
 
   Directory get hostAppRoot;
 
   /// The default 'Info.plist' file of the host app. The developer can change this location in Xcode.
-  File get defaultHostInfoPlist => hostAppRoot.childDirectory(_hostAppProjectName).childFile('Info.plist');
+  File get defaultHostInfoPlist => hostAppRoot.childDirectory(hostAppProjectName).childFile('Info.plist');
```

Comment:
This directory is still `Runner` though, right? That wouldn't necessarily be changed.  When I tested a `xcodeproj`  rename it was still at `Runner`.